### PR TITLE
Update reservoirs

### PR DESCRIPTION
**Pull request by limnoliver**
This is the PR that updates all reservoir data for rerunning GLM models and publishing new data releases. This is mostly just a repull/rerun after adding new data from the national flow and temperature pipelines. However, some code changes included:

https://github.com/USGS-R/delaware-model-prep/pull/118/commits/2e32fef08af21caa6be6790ac0e05735525b37bf This commit modified the pull of releases to call dv instead of uv, which required handling the order in which releases and diversions were being returned from NWIS.

https://github.com/USGS-R/delaware-model-prep/pull/118/commits/e911bc86ac3646303ecc82a73c26ddea69b7f6de This adds a releases by type file that Hayley needs for modeling. 

https://github.com/USGS-R/delaware-model-prep/pull/118/commits/ad773af39ca611d7e6f451de57ca58c1fd0caf22 This adds a diversion-only file that Hayley needs for modeling. This was previously pulled from the interpolated, full water budget dataset, but we don't need to republish that data so I'm pulling it out here.

@hcorson-dosch - I think you could restrict your review to those three commits as they are the biggest code changes. 

@lekoenig - I'm tagging you for review just so you can see what changes happen with changes to the data/code are made in scipiper. Note, a new indicator file (with new hash) is committed if the data change, and new build file gets committed when something gets rebuild (independent of whether or not the file changed in the rebuild). Note in https://github.com/USGS-R/delaware-model-prep/pull/118/commits/dd2c95c0dfb3484032c05b189ee1985be661ed82 I had to add some targets in the `getters.yml` file to complete your code updates.

### CODE REVIEW

**Comment by limnoliver** [comment]
@hcorson-dosch - can you also review 

Also updated DWALLIN predictions after Hayley reran GLM. Here's a comparison of the old and new decay rates for Cannonsville:
image
and Pepacton:
image

And a comparison of test periods across all models: 
image

File: 3_predictions.yml [hunk 1]
```diff
@@ -124,8 +55,8 @@ targets:
   # get GLM reservoir predictions
   3_predictions/in/reservoir_downstream_preds.csv.ind:
     command: get_sb_item(out_ind = target_name,
-      sb_id = I('5f6a28a782ce38aaa2449137'),
-      sb_names = I('reservoir_downstream_preds.csv'))
+      sb_id = I('624f21fad34e21f82769a7b1'),
+      sb_names = I('downstream_preds.csv'))
```
Comment (hcorson-dosch) [inline 3_predictions.yml:59]: Looks like the file is still named `'reservoir_downstream_preds.csv'` on that new sb_id:
image

File: 3_predictions/src/nc_to_feather.R [hunk 2]
```diff
@@ -0,0 +1,12 @@
+extract_sntemp_nc <- function(in_ind, out_ind, firstdate) {
+
+  first_date = as.Date(firstdate) # first model date, used to calculate valid dates below
+
+  out = tidync::tidync(sc_retrieve(in_ind)) %>%
+    tidync::hyper_tibble() %>%
```
Comment (hcorson-dosch) [inline 3_predictions/src/nc_to_feather.R:6]: Ooh I've not tried out `tidync` - will have to do so!

File: getters.yml [hunk 3]
```diff
@@ -41,14 +41,19 @@ targets:
 
 
   #### 2_observations ####
-  # temperature sites (placed on drive from/by ____)
+  # temperature sites and data (placed on drive from/by 2wp-temperature-obs by SKO on 22/04/03)
+  # and verified with gd_confirm_posted
```
Comment (hcorson-dosch) [inline getters.yml:45]: Nice to have this documented!

**Review by hcorson-dosch — APPROVED** [review]
All of these code changes look good, Sam! The only thing that caught my eye is that the filename for the downstream preds on SB doesn't match that defined in `'3_predictions,yml'`

File: 3_predictions.yml (hunk 1 above)
Comment (limnoliver) [reply 3_predictions.yml:59]: oh yes, thanks for pointing this out. I originally had it as downstream_preds but then changed it in the data release repo to be the same as last year.

File: 2_observations/src/data_munge_functions.R [hunk 4]
```diff
@@ -236,20 +236,50 @@ clean_release_dat <- function(in_ind, out_ind, mgd_to_cms) {
 }
 
 # get most recent reservoir release data
-get_releases <- function(out_ind, site_ids, reservoir_names) {
+get_releases <- function(out_ind, site_ids, reservoir_names, cfs_to_cms = 0.0283) {
```
[the remark this answers is not in the document]
Comment (limnoliver) [reply 2_observations/src/data_munge_functions.R:239]: Create an issue.

File: 2_observations/src/data_munge_functions.R [hunk 5]
```diff
@@ -236,20 +236,50 @@ clean_release_dat <- function(in_ind, out_ind, mgd_to_cms) {
...
-    group_by(site_no, date) %>%
-    summarize(release_volume_cms = round(mean(releases_cms, na.rm = TRUE), 2)) %>%
-    ungroup() %>%
-    left_join(sites)
 
-  saveRDS(dat, file = as_data_file(out_ind))
+
+  # in DV, all three reservoirs have columns that represent spillway releases,
+  # controlled releases, and diversions. The numbers are returned as numbered columns,
+  # and all three have different orders that you can't ascertain from the naming convention
```
[the remark this answers is not in the document]
Comment (limnoliver) [reply 2_observations/src/data_munge_functions.R:247]: totally.

**Pull request merged**